### PR TITLE
Fix handling of user-provided default schema

### LINKED ISSUE #18428: SqlStatement#setSchema has no effect on execution
I have a Jet instance where I'm creating an `IMap` called `person` with one record:
```
JetInstance jetInstance = Jet.newJetInstance();
IMap<Object, Object> person = jetInstance.getMap("person");
person.put(1, new Person("public", 1));
```
and a mapping named `person` with one record:
```
SqlService sql = jetInstance.getSql();
sql.execute("CREATE OR REPLACE MAPPING person(__key INT, name VARCHAR, age INT) TYPE IMap " +
                "OPTIONS('keyFormat'='int', 'valueFormat'='json')");
sql.execute("SINK INTO person VALUES(1, 'partitioned', 2)");
```
I expect that `IMap` will be located under schema `public` while the mapping under `partitioned`.

However, when executing the SQL statement with setting different schema it always fetches the results from the `partitioned.person` (mapping):

```
SqlResult res1 = sql.execute(new SqlStatement("SELECT * FROM person").setSchema("partitioned"));
//res1 -> [__key INTEGER=1, name VARCHAR=partitioned, age INTEGER=2]

SqlResult res2 = sql.execute(new SqlStatement("SELECT * FROM person").setSchema("public"));
//res2 -> [__key INTEGER=1, name VARCHAR=partitioned, age INTEGER=2]
```

### PR DESCRIPTION
**Pull request by viliam-durina**
Fixes #18428

The PR lacks a test because it's not possible to test it in IMDG because IMDG currently uses only a single schema (`partitioned`) and any value here is ignored.

A test could be added to Jet, but on the other hand, Jet doesn't yet depend on imdg 5.0. But I tested it locally. ¯\_(ツ)_/¯

### CODE REVIEW

**Comment by gierlachg** [comment]
I would either postpone it or create separate issue for the test so it does not get forbidden.

**Review by gierlachg — APPROVED** [review]

**Comment by viliam-durina** [comment]
@gierlachg I created an issue for that: hazelcast/hazelcast-jet#3002

File: hazelcast-sql-core/src/test/java/com/hazelcast/sql/Testt.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+package com.hazelcast.sql;
+
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import org.junit.Test;
+
+import java.io.Serializable;
+
+public class Testt {
```
Comment (gierlachg) [inline hazelcast-sql-core/src/test/java/com/hazelcast/sql/Testt.java:25]: Give it a more meaningful name?

File: hazelcast-sql-core/src/test/java/com/hazelcast/sql/Testt.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        inst.getMap("m").put(1, new Foo());
+        for (SqlRow r : inst.getSql().execute("select * from m")) {
+            System.out.println(r);
+        }
+    }
+
+    private static class Foo implements Serializable {
+//        public String getField() {
+//            return "1";
+//        }
```
Comment (gierlachg) [inline hazelcast-sql-core/src/test/java/com/hazelcast/sql/Testt.java:39]: Remove?

File: hazelcast-sql-core/src/test/java/com/hazelcast/sql/Testt.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import org.junit.Test;
+
+import java.io.Serializable;
+
+public class Testt {
+
+    @Test
+    public void test() {
```
Comment (gierlachg) [inline hazelcast-sql-core/src/test/java/com/hazelcast/sql/Testt.java:28]: What exactly does it test?

File: hazelcast-sql-core/src/test/java/com/hazelcast/sql/Testt.java (hunk 1 above)
Comment (viliam-durina) [reply hazelcast-sql-core/src/test/java/com/hazelcast/sql/Testt.java:25]: This was included in the commit by mistake, I always use `Testt` for ad-hoc tests :). I'll remove

**Review by Fly-Style — APPROVED** [review]

**Pull request merged**